packages/build/src/plugins_core/blobs_upload/index.ts

```diff
@@ -0,0 +1,62 @@
+import { getDeployStore } from '@netlify/blobs'
+import pMap from 'p-map'
+
+import { addErrorInfo } from '../../error/info.js'
+import { log, logError } from '../../log/logger.js'
+
+import { getKeysToUpload, getBlobsDir, getFileWithMetadata, anyBlobsToUpload } from './utils.js'
+
+const coreStep = async function ({
+  debug,
+  logs,
+  deployId,
+  buildDir,
+  constants: { PUBLISH_DIR, SITE_ID, NETLIFY_API_TOKEN, API_URL },
+}) {
+  const blobStore = getDeployStore({ siteID: SITE_ID, deployID: deployId, token: NETLIFY_API_TOKEN, apiURL: API_URL })
+  const blobsDir = getBlobsDir({ buildDir, publishDir: PUBLISH_DIR })
+  const keys = await getKeysToUpload(blobsDir)
+
+  // We checked earlier, but let's be extra safe
+  if (keys.length === 0) {
+    log(logs, 'No blobs to upload to deploy store.')
+    return {}
+  }
+
+  log(logs, `Uploading ${keys.length} blobs to deploy store...`)
+
+  const uploadBlob = async (key) => {
+    if (debug) {
+      log(logs, `- Uploading blob ${key}`, { indent: true })
+    }
+    const { data, metadata } = await getFileWithMetadata(blobsDir, key)
+    await blobStore.set(key, data, { metadata })
+  }
+
+  try {
+    await pMap(keys, uploadBlob, { concurrency: 10 })
```

Comment:
FYI, the concurrency limit we currently use in the CLI is 5: https://github.com/netlify/cli/blob/5c61169d43570ffb20cfd2976ca050a9558d9c3f/src/utils/deploy/constants.ts#L6.

I don't have any thoughts on which value we should use, just wanted to surface what we're doing in the CLI.

Reply:
Oh good catch. From buildbot we do 10, so I think I had that one in mind. I think we should be OK keeping it at 10 (it's a much simpler upload than a file upload), but I was planning on keeping an eye on buildbot traces anyway.